### PR TITLE
MRG: Updates for latest MNE

### PR DESCRIPTION
**Pull request by Eric89GXL**
Some minor fixes for latest MNE.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@kambysese do these work for you, too?

**Comment by kambysese** [comment]
With the exception of `fir_design` kwarg, yes the updates have worked for me through `write_epochs`

File: mnefun/_mnefun.py [hunk 1]
```diff
@@ -414,9 +416,22 @@ def convert_subjects(self, subj_template, struc_template=None):
...
+            assert all(isinstance(subj, string_types) for subj in new)
+            self.structurals = new
+        if isinstance(subj_template, string_types):
+            def fun(x):
+                return subj_template % x
+        else:
+            fun = subj_template
+        new = [fun(subj) for subj in self.subjects]
+        assert all(isinstance(subj, string_types) for subj in new)
+        self.subjects = new
```
Comment (kambysese) [inline mnefun/_mnefun.py]: Is this bit for inverse modelling in a template_mri?

Comment (Eric89GXL) [inline mnefun/_mnefun.py]: It allows more flexible mapping between `subj` and `struct` names, e.g. if you need to put in a `subj.upper()` for example

**Comment by Eric89GXL** [comment]
Okay @kambysese this should work with old and new MNE now

File: mnefun/_mnefun.py [hunk 2]
```diff
@@ -1706,6 +1707,21 @@ def _cals(raw):
...
+def _get_fir_kwargs(fir_design):
+    """Get FIR kwargs in backward-compatible way."""
+    fir_kwargs = dict()
+    if fir_design != 'firwin2':
+        # only pass if necessary (backward compat)
+        fir_kwargs.update(fir_design=fir_design)
+    old_kwargs = dict()
+    if 'fir_design' in mne.fixes._get_args(mne.filter.filter_data):
+        old_kwargs.update(fir_design='firwin2')
+    elif len(fir_kwargs):
```
Comment (kambysese) [inline mnefun/_mnefun.py]: What's happening here? Isn't `elif len(fir_kwargs)` always going evaluate as True?

Comment (Eric89GXL) [inline mnefun/_mnefun.py]: No, only if `fir_design != 'firwin2'`

**Comment by kambysese** [comment]
Apart from my question LTGM

File: mnefun/_mnefun.py (hunk 2 above)
Comment (kambysese) [inline mnefun/_mnefun.py]: Even though it's not indented into the `fir_design != 'firwin2' ` block? sorry I don't get it...

Comment (Eric89GXL) [inline mnefun/_mnefun.py]: You'll have to think about the various cases, i.e. the values of `fir_design` and the MNE version, and think about the flow. But in any case, I think it should work okay

**Comment by Eric89GXL** [comment]
Thanks @kambysese. Let me know if I broke anything when you actually use it :)

**Pull request merged**